### PR TITLE
Vtx power osd VTX channel variant

### PR DESCRIPTION
**Pull request by P-I-Engineer**
I've created osd variant for VTX channel that displays only the vtx power.

pull request is partial approved in betaflight here
https://github.com/betaflight/betaflight/pull/11275

Screenshot 2022-01-13 100917

Screenshot 2022-01-13 101003

Screenshot 2022-01-13 100949

### CODE REVIEW

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by P-I-Engineer** [comment]
i'm not sure of the process here, am i to add those labels? is there a guide?
thanks

**Review by haslinghuis — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:4863]: To be in line with other variants:
```suggestion
        "description": "One of the variants of the altitude element of the OSD"        
```

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:4866]: Add the description to be aligned with the other. The description is used by the translators.
```suggestion
        "message": "Band:Channel:Pwr:Pit",
        "description": "One of the variants of the altitude element of the OSD"
```

File: src/js/tabs/osd.js
Comment (McGiverGim) [inline src/js/tabs/osd.js:315]: You dont need to use ` you can use ' or " if it isn't a template.
```suggestion
            value = 'R:2:200:P';
```

File: src/js/tabs/osd.js
Comment (McGiverGim) [inline src/js/tabs/osd.js:319]: The same.
```suggestion
            value = 'PWR';
```

**Review by McGiverGim — CHANGES_REQUESTED** [review]
Minor changes only ;)

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by McGiverGim — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> FAIL
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by P-I-Engineer** [comment]
rebase and squash newb issues. it's down to 1 commit now, but has others in it. Any suggestions? or is this ok?

**Comment by P-I-Engineer** [comment]
all fixed now i believe. Should be ready for review. thank you

**Review by haslinghuis — DISMISSED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

File: src/js/tabs/osd.js
Comment (McGiverGim) [inline src/js/tabs/osd.js:586]: Sorry to not notice that the first time...
The other "variants" have not a '-' between the name and the variant. Is good to continue with the same nomenclature and not merge ways.
Another option is to replace all the others, but is simpler to modify only this.

Comment (P-I-Engineer) [reply src/js/tabs/osd.js]: all fixed.

thanks for feedback

**Review by haslinghuis — APPROVED** [review]

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:4810]: To be camelCase need the P uppercase:
```suggestion
    "osdTextElementVTXchannelVariantPower": {
```

File: locales/en/messages.json
Comment (McGiverGim) [inline locales/en/messages.json:4814]: The same here:
```suggestion
    "osdTextElementVTXchannelVariantFull": {
```

File: src/js/tabs/osd.js
Comment (McGiverGim) [inline src/js/tabs/osd.js:586]: And then:
```suggestion
                'osdTextElementVTXchannelVariantFull',
                'osdTextElementVTXchannelVariantPower',
```

**Review by McGiverGim — CHANGES_REQUESTED** [review]

**Comment by P-I-Engineer** [comment]
capitalization fixed

**Review by McGiverGim — APPROVED** [review]
Perfect! Thanks!

**Review by haslinghuis — APPROVED** [review]

**Comment by P-I-Engineer** [comment]
what's next? is there anything i need to do?

**Comment by haslinghuis** [comment]
@P-I-Engineer Thanks.

We are in release state (meaning code freeze - only bug fixes allowed) for some time now. As this is a new feature it has to wait for the next version.

**Comment by P-I-Engineer** [comment]
ok, sounds good, thanks for info.

File: src/js/tabs/osd.js
Comment (limonspb) [inline src/js/tabs/osd.js:319]: probably instead of PWR it should show, for example: "250MW" or whatever format is correct.

**Review by limonspb — CHANGES_REQUESTED** [review]

File: src/js/tabs/osd.js
Comment (P-I-Engineer) [reply src/js/tabs/osd.js:319]: I have it as "PWR" because the displayed value is just a number without a unit to keep the OSD element as short as possible. I can change to "200". 

thoughts?

Comment (limonspb) [reply src/js/tabs/osd.js:319]: ok yeah i think in this case its ok

**Review by limonspb — DISMISSED** [review]

File: src/js/tabs/osd.js
Comment (McGiverGim) [reply src/js/tabs/osd.js:319]: We usually use one of the "real" representations in the OSD. If it is simply a number, use a number.

**Review by McGiverGim — DISMISSED** [review]

**Review by limonspb — CHANGES_REQUESTED** [review]
Please make sure you always have one commit in the PR so it's ready to merge

File: .nvmrc
Comment (haslinghuis) [inline .nvmrc:5]: Please remove .nvmrc changes.

Please look here https://github.com/betaflight/betaflight/wiki/Betaflight-Contribution#squash-your-commits how to squash your commits

**Comment by haslinghuis** [comment]
@P-I-Engineer you might need: https://github.com/betaflight/betaflight/wiki/Betaflight-Contribution#update-your-local-branch-with-upstream-changes before doing the squash as outlined earlier.

File: .nvmrc
Comment (P-I-Engineer) [reply .nvmrc]: i'm battling github right now to figure out how to squash and stop messing stuff up

Comment (haslinghuis) [reply .nvmrc]: Been there done that. Spent hours to get a grasp. So hope my document will guide any developers as these are my working notes.

Comment (haslinghuis) [reply .nvmrc]: Actually days 😂

**Comment by P-I-Engineer** [comment]
closing, too many github issues to resolve.

see new PR https://github.com/betaflight/betaflight-configurator/pull/2964

**Pull request closed without merging**
